systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -0,0 +1,184 @@
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Tag(REGRESSION)
+public class KafkaNodePoolST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(KafkaNodePoolST.class);
+
+
+    /**
+     * @description This test case verifies transfer of Kafka Cluster from and to management by KafkaNodePool, by creating corresponding Kafka and KafkaNodePool custom resources
+     * and manipulating according kafka annotation.
+     * @steps 1. - Deploy Kafka with annotated to enable management by KafkaNodePool, and KafkaNodePool targeting given Kafka Cluster.
+     * - Kafka is deployed, KafkaNodePool custom resource is targeting Kafka Cluster as expected.
+     * 2. - Modify KafkaNodePool by increasing number of Kafka Replicas.
+     * - Number of Kafka Pods is increased to match specification from KafkaNodePool
+     * 3. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     * 4. - Modify Kafka custom resource annotation strimzi.io/node-pool to disable management by KafkaNodePool.
+     * - new StatefulSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) are ignored.
+     * 5. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     * 6. - Modify Kafka custom resource annotation strimzi.io/node-pool to enable management by KafkaNodePool.
+     * - new StatefulSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) has priority over Kafka specification.
+     * 7. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     *
+     * @usecase
+     * - kafka-node-pool
+     */
+    @ParallelNamespaceTest
+    void testKafkaManagementTransferToAndFromKafkaNodePool(ExtensionContext extensionContext) {
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        final int originalKafkaReplicaCount = 3;
+        final int nodePoolIncreasedKafkaReplicaCount = 4;
+
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(200)
+            .withNamespaceName(testStorage.getNamespaceName())
+            .build();
+
+        LOGGER.info("Deploying Kafka Cluster: {}/{} controlled by KafkaNodePool: {}", testStorage.getNamespaceName(), testStorage.getClusterName(), testStorage.getKafkaNodePoolName());
```

Comment:
What is the `testStorage.getKafkaNodePoolName()` here?

Reply:
it refers to the name of KafkaNodePool CR which will be created in a a moment which will be targeting given Kafka CR

Reply:
So, what exactly are you trying to test in this test? It seems from the name and description that you want to test the migration to and from node pools. Which is a good test. But it has to follow the process how it should be done: https://strimzi.io/docs/operators/in-development/full/deploying.html#proc-migrating-clusters-node-pools-str ... the node pool is always `kafka` when migrating there or back. It should not have some random name.

Reply:
1.) test starts with Cluster Operator with fature gate +kafkaNodePool (at the moment it does not change during  the test)

2.) Kafka CR  and KafkaNodePool CR are created, Kafka CR having annotation which enables it to be controlled by KafkaNodePools, and  KafkaNodePool having label which points to given Kafka.

3.)  as Kafka is now  "managed" by KafkaNodePool we try to modify KafkaNodePool and observe these changes are propagated. 

4. 5.) : we  disable and enable KafkaNodePools (only) by manipulating Annotation in Kafka CR i.e., 
`strimzi.io/node-pools` enabled/disabled. which is almost like in given documentation except for the fact that feature gate in Cluster Operator is active whole the time. 

Would you like to be disabled as part of this test as well ?

Reply:
I think it should be disabled. But the point I was asking about was the name of the Node Pool. Is it named `kafka` or not?

Reply:
the name is for example `kafka-pool-451e36b2` i.e. kafka-pool- + som random suffix

Reply:
Right, then it does not do the migration and the test IMHO makes not much sense.

Reply:
understand, changing to "kafka".